sotodlib/tod_ops/gapfill.py

```diff
@@ -380,3 +383,69 @@ def get_contaminated_ranges(good_flags, bad_flags):
                 rs.add_interval(int(i0), int(i1))
     return contam
 
+
+def fill_glitches(aman, nbuf=10, use_pca=False, modes=3, signal=None,
+                  glitch_flags=None, merge_filled=True,
+                  filled_name='gap_filled'):
+    """
+    Top level function for filling glitches assuming glitch flags exist.
+    Wraps the other functions in the ``tod_ops.gapfill`` module.
+
+    Args
+    -----
+    aman : AxisManager
+        AxisManager to fill glitches in
+    nbuf : int
+        Number of buffer samples to use in polynomial gap filling.
+    use_pca : bool
+        Whether or not to fill glitches using pca model. Default is False
+    modes : int
+        Number of modes in the pca to use if pca=True. Default is 3.
+    signal : ndarray or None
+        Array of data to fill glitches in. If None then uses ``aman.signal``.
+        Default is None.
+    glitch_flags : RangesMatrix or None
+        RangesMatrix containing flags to use for gap filling. If None then
+        uses ``aman.flags.glitches``.
+    merge_filled : bool
+        Merge gap filled data into input axis manager.
+    filled_name : str
+        Axis name to merge gap filled data if merge_filled is True.
+
+    Returns
+    -------
+    signal : ndarray
+        Returns ndarray with gaps filled from input signal.
+    """
+    if signal is None:
+        sig = np.copy(aman.signal)
+    else:
+        sig = np.copy(signal)
+    
+    if glitch_flags is None:
+        glitch_flags = aman.flags.glitches
+    if use_pca is True:
+        if modes > aman.dets.count:
+            logger.warning(f'modes = {modes} > number of detectors = ' +
+                           f'{aman.dets.count}, setting modes = number of ' +
+                           'detectors')
+            modes = aman.dets.count
+        # fill with poly fill before PCA
+        gaps = get_gap_fill(aman, nbuf=nbuf, flags=glitch_flags,
+                            signal=np.float32(sig))
```

Comment:
It is annoying this needs to be converted to float32 ... will create an issue to resolve that.